src/main/java/org/spongepowered/api/CatalogType.java

```diff
@@ -25,24 +25,41 @@
 package org.spongepowered.api;
 
 /**
- * Represents a type of a catalog that can be used to identify types
- * without using an {@link Enum}.
+ * Represents a type of a catalog that can be used to identify types without
+ * using an {@link Enum}.
+ *
+ * <p>Implementing classes must meet the following requirements for all its
+ * members, including those which are not listed in the classes listed in the
+ * {@link org.spongepowered.api.util.annotation.CatalogedBy CatalogedBy}
+ * annotation:
+ * <ul>
+ * <li>Every pair (a, b) that matches:
```

Comment:
I feel like its a little easier to simply say 'if any of `a.equals(b)`, `a == b`, or `a.getId().equalsIgnoreCase(b.getId())` are true then all must be true' rather than the nested lists. So something like:

```
Implementing classes must meet the requirement for all its members that if any of `a.equals(b)`, `a == b`, or 
`a.getId().equalsIgnoreCase(b.getId())` are true then all must be true. This must also be also applies to members 
which are not listed in the classes listed in the {@link org.spongepowered.api.util.annotation.CatalogedBy 
CatalogedBy} annotation but nevertheless implement this interface.
```